frontend/app/src/components/calendar/CalendarGrid.vue

```diff
@@ -0,0 +1,132 @@
+<script setup lang="ts">
+import dayjs, { type Dayjs } from 'dayjs';
+import type { CalendarEvent } from '@/types/history/calendar';
+
+const props = defineProps<{
+  today: Dayjs;
+  selectedDate: Dayjs;
+  eventsWithDate: (CalendarEvent & { date: string })[];
+}>();
+
+const emit = defineEmits<{
+  (e: 'update:selected-date', selectedDate: Dayjs): void;
+  (e: 'update:range', range: [number, number]): void;
+  (e: 'edit', event: CalendarEvent): void;
+}>();
+
+function setSelectedDate(day: Dayjs) {
+  emit('update:selected-date', day);
+}
+
+function edit(event: CalendarEvent) {
+  emit('edit', event);
+}
+
+const { today, selectedDate, eventsWithDate } = toRefs(props);
+
+const month = computed(() => Number(get(selectedDate).format('M')));
+const year = computed(() => Number(get(selectedDate).format('YYYY')));
+
+const numberOfDaysInMonth: ComputedRef<number> = computed(() => dayjs(get(selectedDate)).daysInMonth());
+
+function getWeekday(date: string | Dayjs) {
+  const dayjsValue = typeof date === 'string' ? dayjs(date) : date;
+  return dayjsValue.weekday();
+}
+
+const currentMonthDays = computed(() => [...new Array(get(numberOfDaysInMonth))].map((_, index) => ({
+  date: dayjs(`${get(year)}-${get(month)}-${index + 1}`),
+  isCurrentMonth: true,
+})));
+
+const previousMonthDays = computed(() => {
+  const firstDayOfTheMonthWeekday = getWeekday(
+    get(currentMonthDays)[0].date,
+  );
+  const previousMonth = dayjs(`${get(year)}-${get(month)}-01`).subtract(
+    1,
+    'month',
+  );
+
+  // Cover first day of the month being sunday (firstDayOfTheMonthWeekday === 0)
+  const visibleNumberOfDaysFromPreviousMonth = firstDayOfTheMonthWeekday
+    ? firstDayOfTheMonthWeekday - 1
+    : 6;
+
+  const previousMonthLastMondayDayOfMonth = dayjs(
+    get(currentMonthDays)[0].date,
+  )
+    .subtract(visibleNumberOfDaysFromPreviousMonth, 'day')
+    .date();
```

Comment:
Any chance your editor is still configured to 80 line length @lukicenturi?
if so can you switch it to 120?